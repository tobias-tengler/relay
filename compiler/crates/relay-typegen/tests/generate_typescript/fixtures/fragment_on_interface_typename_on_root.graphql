fragment a on Node {
  __typename
  ... on User {
    name
  }
  ... on Page {
    doesViewerLike
  }
}