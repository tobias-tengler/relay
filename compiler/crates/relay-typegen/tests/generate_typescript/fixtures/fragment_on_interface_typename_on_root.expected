==================================== INPUT ====================================
fragment a on Node {
  __typename
  ... on User {
    name
  }
  ... on Page {
    doesViewerLike
  }
}
==================================== OUTPUT ===================================
import { FragmentRefs } from "relay-runtime";
export type a$data = {
  readonly __typename: "Page";
  readonly doesViewerLike: boolean | null | undefined;
  readonly " $fragmentType": "a";
} | {
  readonly __typename: "User";
  readonly name: string | null | undefined;
  readonly " $fragmentType": "a";
} | {
  // This will never be '%other', but we need some
  // value in case none of the concrete values match.
  readonly __typename: "%other";
  readonly " $fragmentType": "a";
};
export type a$key = {
  readonly " $data"?: a$data;
  readonly " $fragmentSpreads": FragmentRefs<"a">;
};
